tests/test_army_firearms.py

```
@@ -92,71 +99,114 @@ def test_create_army_with_firearm(game):
+    assert name in rpt_a and "6" in rpt_a
+    assert rpt_a != rpt_b
 
+    # roster 双态：数值火器 vs 定性火器；炮门数两态皆在
+    db.conn.execute(
+        "UPDATE armies SET firearm_equipment=30, cannon_equipment=4 WHERE id=?",
+        (aid,),
+    )
+    db.conn.commit()
+    line_num = next(
+        l
+        for l in db.army_roster(filter_names=[name], qualitative_equipment=False).splitlines()
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)